src/test/python/replay_target.py

```diff
@@ -16,516 +27,56 @@ def replay_target(target):
         target.modify_gradle(sdk_version)
 
     # server start
-    docker_up_cmd = "cd " + target.test_repo_name + " && docker compose -f conf/docker-compose.yml up -d"
-    val_1 = os.system(docker_up_cmd)
-
-    proc = subprocess.Popen(["docker ps -a"], stdout=subprocess.PIPE, shell=True)
-    (out_stream, err_stream) = proc.communicate()
-    docker_container_name = "target-repo"
+    if target.target_run_properties.start_mode == StartMode.DOCKER:
+        docker_up_cmd = "cd " + target.test_repo_name + " && docker compose -f conf/docker-compose.yml up -d"
+        os.system(docker_up_cmd)
 
-    # target replay
-    if (target.buildSystem == buildSystem.MAVEN):
-        test_command = "docker exec " + docker_container_name + " ./mvnw clean install surefire:test --fail-never"
-    elif (target.buildSystem == buildSystem.GRADLE):
-        test_command = "docker exec " + docker_container_name + " ./gradlew test"
-    val_2 = os.system(test_command)
+        proc = subprocess.Popen(["docker ps -a"], stdout=subprocess.PIPE, shell=True)
+        (out_stream, err_stream) = proc.communicate()
+        docker_container_name = "target-repo"
 
-    response_code = val_1 or val_2
-    if (response_code == 0):
-        print("Test for target executed successfully: " + target.test_repo_name)
+        # target replay
+        if (target.buildSystem == buildSystem.MAVEN):
+            test_command = "docker exec " + docker_container_name + " ./mvnw clean install surefire:test --fail-never"
+        elif (target.buildSystem == buildSystem.GRADLE):
+            test_command = "docker exec " + docker_container_name + " ./gradlew clean test"
+        os.system(test_command)
     else:
-        raise Exception("Test for target failed execution: " + target.test_repo_name)
+        if (target.buildSystem == buildSystem.MAVEN):
+            test_command = "cd "+target.test_repo_name+" && ./mvnw clean test"
+        elif (target.buildSystem == buildSystem.GRADLE):
+            test_command = "cd "+target.test_repo_name+" && ./gradlew clean test"
+        os.system(test_command)
+
```

Comment:
Why are we calling different commands for docker and CMD run types in case of maven build systems?

Reply:
./mvnw clean test would generally not make a surefire report. So the parsing after this will not work. Is this flow tested (cmd run and maven)?

Reply:
The `./mvnw clean test` is indeed creating a surefire xml file. Then why are we running a different command for docker runs with surefire-report?
Basically why run `./mvnw clean install surefire:test --fail-never` over `./mvnw clean test` for docker envs?